test/application/sanitizer/sanitizer.spec.js

```diff
@@ -43,6 +43,13 @@ describe('sanitizer test suite', function () {
 
       expect(sanitize(html)).toBe(expectedOutput);
     });
+
+    it('dangerous instructions in the href attribute', function () {
+      const html = '<section><h1>Testing</h1><div><a href="javascript:alert(\'XSS\');">Yo</a></div></section>';
```

Comment:
I think we need a check to make sure we sanitize text that starts with a white list item but holds dangerous instructions still.

maybe these cases too:
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#Default_SRC_tag_to_get_past_filters_that_check_SRC_domain
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#On_error_alert

Also making sure the file is not JS:
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#Non-alpha-non-digit_XSS
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#No_closing_script_tags
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#Double_open_angle_brackets
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#Remote_style_sheet

Wondering if this would also be passing in something different than a `meta` tag:
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#META_with_additional_URL_parameter

https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#IFRAME_Event_based
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#Using_an_EMBED_tag_you_can_embed_a_Flash_movie_that_contains_XSS
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#Assuming_you_can_only_fit_in_a_few_characters_and_it_filters_against_.22.js.22

Where it gets also a bit more complicated to check:
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#IMG_Embedded_commands

😱 
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#IMG_Embedded_commands_part_II
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#XSS_using_HTML_quote_encapsulation

This one looks actually problematic:
https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#URL_string_evasion

Reply:
From my understanding of how this library works, a whitelist attrtibute doesn’t mean it’s clear to pass any logic. Because `download` was not in the whitelist, it would be stripped out completely. But even then the attributes in the whitelisted `href` tag would still be processed and checked if it had allowable links. 

So anything not whitelisted is auto stripped out, from my understanding.

Reply:
yeah, just making sure what we want passes and some for what we don't want, the 3 that really concern me are:
- https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#IMG_Embedded_commands
- https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#IMG_Embedded_commands_part_II
- https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#URL_string_evasion

I am not sure we can solve that, unless we have a whitelist of hosts too.

Reply:
I added some more tests, but I don't see how any of these relates to my PR specifically? Img embedding has nothing to do with the `a` attribute and url evasion only matters when specific domains are being blocked.